File: frontend/src/types/user-admin.ts
import type { UserRole } from "@/types/api";
import type { UserSummary } from "@/features/auth/types";

export type UserStatusFilter = "active" | "inactive" | "all";

// Strictly wider than UserSummary (the colleague directory's shape): only the
// admin endpoints return isActive and the deactivation audit pair, which is
// why this lives in its own type rather than extending the shared one.
export interface AdminUser {
  id: string;
  displayName: string;
  email: string;
  role: UserRole;
  isActive: boolean;
  createdAt: string;
  deactivatedAt: string | null;
  deactivatedBy: UserSummary | null;
}

export interface PagedUsers {
  items: AdminUser[];
  page: number;
  size: number;
  total: number;
  totalPages: number;
}

File: backend/app/schemas/user_admin.py
import uuid
from datetime import datetime

from app.db.models.enums import UserRole
from app.schemas.auth import UserSummary
from app.schemas.base import CamelModel


class UserAdminSummary(CamelModel):
    """The admin view of a user. Strictly wider than UserSummary — it exposes
    is_active and the deactivation audit pair, which is exactly why the
    colleague directory keeps returning the narrower one."""

    id: uuid.UUID
    display_name: str
    email: str
    role: UserRole
    is_active: bool
    created_at: datetime
    deactivated_at: datetime | None = None
    deactivated_by: UserSummary | None = None


class PagedUsers(CamelModel):
    items: list[UserAdminSummary]
    page: int
    size: int
    total: int
    total_pages: int


class UserRoleUpdate(CamelModel):
    role: UserRole
